### PR TITLE
Add GHA for Linting

### PR DESCRIPTION
**Pull request by fkneeland-figure**
## Description

closes: #61 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/explorer/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by fkneeland-figure** [comment]
@crussell-figure is this what you were thinking of?  I pulled this from https://ktlint.github.io/#getting-started and https://github.com/JLLeitschuh/ktlint-gradle

**Comment by fkneeland-figure** [comment]
With this change you can run: ./gradlew ktlintCheck

Do we want to run that in CI or on commit as well?

File: service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt [hunk 1]
```diff
@@ -153,10 +156,12 @@ class BlockProposerRecord(id: EntityID<Int>) : IntEntity(id) {
...
-                this.blockTimestamp = timestamp!!
-            }).apply {
+            (
+                BlockProposerRecord.findById(height) ?: new(height) {
+                    this.proposerOperatorAddress = proposer!!
+                    this.blockTimestamp = timestamp!!
+                }
+                ).apply {
                 this.minGasFee = minGasFee
             }
```
Comment (crussell-figure) [inline service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt:166]: This makes me a sad person...

Comment (egaxhaj-figure) [reply service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt:166]: Absolutely! Is this lint's recommendation? If so, maybe there's a way to ignore it

**Review by egaxhaj-figure — DISMISSED** [review]
I added two recommendations we can get around lint's recommendation in [Blocks.kt](service/src/main/kotlin/io/provenance/explorer/domain/entities/Blocks.kt)

**Pull request merged**